### PR TITLE
nukie gun: spes-12

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

additional sound by grungus

adds new nukie gun: spes-12
additionally adds a "tactical bandolier", a 30-round waist bandolier to go with it

the spes-12 is a shotgun that holds 8+1 casings (1 chambered) with two fire modes: semi-auto and pump-action
in pump-action the spread variance of pellets is halved and any other projectile is more accurate
in semi-auto you get none of these benefits but it is semi-auto
it also kinda hurts to get hit with and you may put it in your suit storage on your syndicate modsuit so you can carry it with the tactical bandolier equipped

image
image

https://github.com/user-attachments/assets/d1638e5b-91ed-4069-8e51-6a351379cda9

## Why It's Good For The Game
the SPES-12 is better for more accurate gunning in tighter corridors compared to the Bulldog; the pellets hit the walls much less in a tight corridor like maintenance
its also good if youre not bad at aiming like all bulldog users

## Changelog
:cl:
add: added the SPES-12 kit (and ammo) to the nukie uplink, and a belt to go with it
/:cl:

### CODE REVIEW

File: sound/attributions.txt [hunk 1]
```diff
@@ -209,3 +209,6 @@ https://freesound.org/people/C_Rogers/sounds/203368/ -- glass-shattering-hit_01.
 https://freesound.org/people/Czarcazas/sounds/330800/ -- Audio reversal/fading of Shattering Glass (Small) by Czarcazas -- https://freesound.org/s/330800/ -- License: Attribution 3.0
 
 sound/effects/bonk.ogg - recorded by oranges on a coke zero bottle, edited by ninjanomnom, released to public domain
+shell_insert2.ogg - https://freesound.org/people/CombatSFX4You/sounds/563650/ , License: CC BY-NC 3.0
+shotgun_click1.ogg - https://freesound.org/people/PNMCarrieRailfan/sounds/681694/ , License: CC BY-NC 4.0
+shotgun_fire1.ogg - https://freesound.org/people/michorvath/sounds/427595/ , License: CC0
```
Comment (Sadboysuss) [inline sound/attributions.txt:214]: preferably put the attribution text in the same folder as the sound files, that would make it easier to find the relevant attributions in the future.

**Review by grungussuss — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 2]
```diff
@@ -384,3 +384,66 @@
...
+	force = 13 // have you ever got hit in the face with a shotgun? thankfully havent experienced but probably hurts
+	projectile_damage_multiplier = 1.2 // on par with a bulldog
+	can_suppress = FALSE
+	fire_delay = 4
+	pin = /obj/item/firing_pin/implant/pindicate
+	accepted_magazine_type = /obj/item/ammo_box/magazine/internal/shot/spes12
+
+/obj/item/gun/ballistic/shotgun/spes12/examine(mob/user)
+	. = ..()
+	. += "It is currently firing in [semi_auto ? "semi-auto" : "pump-action"] mode. Right-click to toggle mode."
```
Comment (MrMelbert) [inline code/modules/projectiles/guns/ballistic/shotgun.dm:407]: Screentip?

File: code/game/objects/items/storage/belt.dm [hunk 3]
```diff
@@ -748,6 +748,39 @@
...
+	desc = "Tactical waist strap bandolier to hold 30 rounds of shotgun-sized ammo."
+	icon_state = "tactibandolier"
+	worn_icon_state = "tactibandolier"
+
+/obj/item/storage/belt/bandolier/tactical/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+	atom_storage.max_slots = 30
+	atom_storage.max_total_storage = 30
+	atom_storage.do_rustle = FALSE
```
Comment (MrMelbert) [inline code/game/objects/items/storage/belt.dm:762]: This storage should probably be whitelisted to hold ammo only

File: code/game/objects/items/storage/belt.dm [hunk 4]
```diff
@@ -748,6 +748,39 @@
...
+	update_appearance()
+
+/obj/item/storage/belt/bandolier/tactical/update_icon_state() // Looks like an assault belt if empty, otherwise normal icon state
+	if(length(contents))
+		icon_state = initial(icon_state)
+		worn_icon_state = initial(worn_icon_state)
+		return ..()
+	icon_state = /obj/item/storage/belt/military/assault::icon_state
+	worn_icon_state = /obj/item/storage/belt/military/assault::worn_icon_state
+	return ..()
```
Comment (MrMelbert) [inline code/game/objects/items/storage/belt.dm:772]: This is weird, why does it change style when empty

File: code/game/objects/items/storage/belt.dm (hunk 3 above)
Comment (necromanceranne) [reply code/game/objects/items/storage/belt.dm:762]: Wouldn't that already be true since its a bandolier subtype.

Comment (mc-oofert) [reply code/game/objects/items/storage/belt.dm:762]: it is

File: code/game/objects/items/storage/belt.dm (hunk 4 above)
Comment (mc-oofert) [reply code/game/objects/items/storage/belt.dm:772]: so the sprite actually correlates with whether you have any ammo on it unless you want there to be like 30 magic casings on the sprite

**Comment by carlarctg** [comment]
i think it should be called the spess-13

**Comment by carlarctg** [comment]
this can barely be called a spas-12 homage when it really looks just like a Generic Pump Shotgun tbh

if it were up to me id give it a SPAS stock that gets flipped up on pump action and down on semiauto

File: code/game/objects/items/storage/belt.dm (hunk 4 above)
Comment (MrMelbert) [reply code/game/objects/items/storage/belt.dm:772]: Why not just make the ammo an overlay

**Comment by EEASAS** [comment]
i agree this just looks like a stockless pump shotgun

**Comment by deathrobotpunch** [comment]
Bandolier sprite looks ugly

**Comment by DaydreamIQ** [comment]
Being slightly more accurate at the cost of it being pump action seems incredibly underwhelming for a nukie weapon, the bulldog might be more inaccurate but at least you can pump way more rounds into whatever you're missing.

**Comment by necromanceranne** [comment]
Addressing the weapon itself, this doesn't strike me as something that adds much of anything to the uplink that isn't already present. We have a shotgun, it does shotgun stuff, this just seems like an inferior shotgun by comparison. Even your pricing for the ammo boxes are a ripoff for TC; a Bulldog buckshot magazine has 8 shells, is small sized and 2 TC. And breacher slugs are priced equivalent to meteor slugs, when meteor slugs blow breachers completely out of the water in terms of power and utility. They're not even comparable; meteor slugs probably win you a fight in a single shot and tear through practically anything they shoot. Walls, structures, airlocks, whatever. They're literally cannonballs.

There is nothing mechanically advantageous to a internal magazine shotgun, since loading a single shell at a time is almost always worse in a fast-paced conflict that nukies will need to engage in compared to tactical reloads. The comparable example of a weapon like that in the uplink is either the grenade launcher on the M-90gl, or the rocket launcher. And they kill people in one shot if it connects. You're not getting that benefit from this gun.

Even the advertised upside is a false benefit, as pellet cones at the ranges demonstrated do comparably lower damage to targets within 3 tiles, where the cone hasn't started fanning out widely enough for the pellets to not hit center mass if you click right on someone. The pump mode also does nothing meaningfully better for slugs.

If the entire reason this would be added is purely for the vibes, then I disagree with it even more. I actively removed redundant weapons from the uplink, because I think any inclusion should be novel and markedly change how you play the role. This does not do that at all. You either use shotgun A or shotgun B. On paper, one of these definitely feels like the wrong choice.

(If anything, I just think maybe shotgun pellets should be tighter as an entire category, it kind of looks nicer)

**Comment by mc-oofert** [comment]
> this can barely be called a spas-12 homage when it really looks just like a Generic Pump Shotgun tbh
> 
> if it were up to me id give it a SPAS stock that gets flipped up on pump action and down on semiauto

I suck at spriting ok

**Comment by mc-oofert** [comment]
> Addressing the weapon itself, this doesn't strike me as something that adds much of anything to the uplink that isn't already present. We have a shotgun, it does shotgun stuff, this just seems like an inferior shotgun by comparison. Even your pricing for the ammo boxes are a ripoff for TC; a Bulldog buckshot magazine has 8 shells, is small sized and 2 TC. And breacher slugs are priced equivalent to meteor slugs, when meteor slugs blow breachers completely out of the water in terms of power and utility. They're not even comparable; meteor slugs probably win you a fight in a single shot and tear through practically anything they shoot. Walls, structures, airlocks, whatever. They're literally cannonballs.
> 
> There is nothing mechanically advantageous to a internal magazine shotgun, since loading a single shell at a time is almost always worse in a fast-paced conflict that nukies will need to engage in compared to tactical reloads. The comparable example of a weapon like that in the uplink is either the grenade launcher on the M-90gl, or the rocket launcher. And they kill people in one shot if it connects. You're not getting that benefit from this gun.
> 
> Even the advertised upside is a false benefit, as pellet cones at the ranges demonstrated do comparably lower damage to targets within 3 tiles, where the cone hasn't started fanning out widely enough for the pellets to not hit center mass if you click right on someone. The pump mode also does nothing meaningfully better for slugs.
> 
> If the entire reason this would be added is purely for the vibes, then I disagree with it even more. I actively removed redundant weapons from the uplink, because I think any inclusion should be novel and markedly change how you play the role. This does not do that at all. You either use shotgun A or shotgun B. On paper, one of these definitely feels like the wrong choice.
> 
> (If anything, I just think maybe shotgun pellets should be tighter as an entire category, it kind of looks nicer)

We gave nuclear operatives a bullshit busted shotgun that someone made to look extremely tacticool and bad so the rest of the uplink has to suffer from what I'm getting here


also ShiftE that's why you get a bandolier

**Comment by Xackii** [comment]
Nukes are Nukes. They already have 3-4 types of firearms that are almost the same (lg, c-20, m-90gl, smg). If they will have two shotguns to choose from and not a one it will be cool. It will be cool also because this shotgun at least looks like a shotgun and not like a weapon from fortnite (seriously old bulldog sprite looked like the shotgun from far cry 1 it was so cool).

**Comment by mc-oofert** [comment]
if anyone of you ever played tgmc reloading a pump shotgun in the middle of combat was peak

**Comment by mc-oofert** [comment]
i could also just make it a surplus nukie gun but it comes with 38 shells

File: code/game/objects/items/storage/belt.dm (hunk 4 above)
Comment (mc-oofert) [reply code/game/objects/items/storage/belt.dm:772]: true

**Comment by necromanceranne** [comment]
> We gave nuclear operatives a bullshit busted shotgun that someone made to look extremely tacticool and bad so the rest of the uplink has to suffer from what I'm getting here also ShiftE that's why you get a bandolier
> 
> all in all: overanalysis its a shotgun, easy to use this thing to murder anything

Okay.

**Comment by Blacklist897** [comment]
honestly this could be a good tot item like the goon spes-12

**Comment by DaCoolBoss** [comment]
Nukie gun balance has been needing redone for a while, and I like the idea of having two of every kind of weapon with slightly different niches. Like we have a 3-round burst carbine and submachine gun, I don't think that's a problem I just think they need to feel and act slightly differently. Bulldog is primarily a crowd-control tool for rapidly emptying a room, a slower shotgun for duelling high priority targets has a solid niche I feel.

I'm thinking this shotgun could use the ability to add torch/bayonet attachments. High force is nice but being able to inflict slashing wounds without switching would make it terrifying in close range.

**Comment by Paxilmaniac** [comment]
Can't believe we're making a whole new gun just because someone doesn't like the sprite of the bulldog.

**Comment by mc-oofert** [comment]
> Bandolier sprite looks ugly

people say this and get into arguments over PRs related to some shitty 3 colors palette 1996 macguffin sprite

**Comment by mc-oofert** [comment]
> Nukie gun balance has been needing redone for a while, and I like the idea of having two of every kind of weapon with slightly different niches. Like we have a 3-round burst carbine and submachine gun, I don't think that's a problem I just think they need to feel and act slightly differently. Bulldog is primarily a crowd-control tool for rapidly emptying a room, a slower shotgun for duelling high priority targets has a solid niche I feel.
> 
> I'm thinking this shotgun could use the ability to add torch/bayonet attachments. High force is nice but being able to inflict slashing wounds without switching would make it terrifying in close range.

will do later

**Comment by mc-oofert** [comment]
> Can't believe we're making a whole new gun just because someone doesn't like the sprite of the bulldog.

to be fair i did this because i felt like it
also uhhh the sprite for bulldogs is a horrible horrible mess of colors that you will have comparably more trouble finding in a messed up room than the prior sprite

**Comment by Loafin34** [comment]
couldn't you name it the Fusilli M4 or something? why you gotta port jokes FROM GOON?

**Comment by mc-oofert** [comment]
> couldn't you name it the Fusilli M4 or something? why you gotta port jokes FROM GOON?

originally i wanted to name it the B-42 Berserker Shotgun from Soldier of Fortune (2000) but this has nothing to do with goon it just happened to be the first name i thought of before that one

**Comment by SmArtKar** [comment]
Do we really need more Goon content? Being just inspired is not a bad thing, but you've consistently been porting goonstation features (stickers, failed artisci, space pods and now this) and it just feels icky. I think that something like a tight choke for bulldog will fit nukies much better, warops (which is the only scenario where you'd really want to use this) gameplay is too dynamic to have a spare second to reload your shotgun shell-by-shell.

**Comment by DaCoolBoss** [comment]
This isn't a goon port, it just has a similar name to a similar item on goon.

**Comment by SmArtKar** [comment]
> This isn't a goon port, it just has a similar name to a similar item on goon.

Its an item with an exactly the same name, extremely similar sprite and behavior coming from a person who's past feature PRs have been almost 1 to 1 ports of goonstation features (including license violating artisci). Port or not, even if its fine by license I do not think that its a very good look nor does it fit nukie gameplay a lot in its current state.

**Comment by carlarctg** [comment]
i couldve sworn it used to be called spacker 12

**Comment by mc-oofert** [comment]
>  extremely similar sprite

mf thats how a spas 12 looks

**Comment by mc-oofert** [comment]
i might retry something else later but that statement is extremely idiotic and i need this energy to work on my other actual projects
also artsci will come back trust

**Comment by Sakamoto4ka** [comment]
Cool range shotgun

**Comment by vendingmachine685** [comment]
shotguns are shit for nukies either way except slugs

**Comment by Xackii** [comment]
Give it smaller spread so it can shoot further and it will be a cool killer gun (also more damage or more damage to objects).

**Pull request closed without merging**